joomla/joomla-cms
# [com_categories] add searchtools to modal layout and solves bugs

#### Description

This PR adds searchtools to com_categories modal layout categories view.

Also, in normal layout:
- correct bug: the modal on mobile xs show a column without title.
- correct bug: ordering the view for Association doesn't add the arrow icon (removed ordering in this column).
- improves responsive visual (mobile/tablet/desktop)
- solves other minor graphical issues.
- minor code style improvements
###### Before PR (modal)

![before-pr](https://cloud.githubusercontent.com/assets/9630530/13304605/8fd463fa-db4d-11e5-8615-5090b526e07c.png)
###### After PR  (modal)

![after-pr](https://cloud.githubusercontent.com/assets/9630530/13304609/9478628a-db4d-11e5-8467-2b5e6843bc87.png)
#### How to test

Normal view:
1. Install latest staging and apply this patch
2. Go to Content -> Categories
3. Check if search, filters, ordering and pagination are working properly
4. Test on mobile and tablet (ex: reduce browser width)

Modal view:
1. Enable language associations.
2. Create some categories in several languages.
3. Edit one of those categories and go to "Associations" tab and click on "Select" on any language.
4. Check if search, filters, ordering and pagination are working properly in modal.
5. Test on mobile and tablet (ex: reduce browser width)

Should be also tested in other extensions that use categories (ex: User Notes, Contacts, Banners and News Feeds)
